mentat/agent_handler.py

```diff
@@ -0,0 +1,155 @@
+    ChatCompletionMessageParam,
+    ChatCompletionSystemMessageParam,
+)
+
+from mentat.code_feature import CodeMessageLevel
+from mentat.llm_api_handler import count_tokens, prompt_tokens
+from mentat.prompts.prompts import read_prompt
+from mentat.session_context import SESSION_CONTEXT
+from mentat.session_input import ask_yes_no, collect_user_input
+from mentat.transcripts import ModelMessage
+
+agent_file_selection_prompt_path = Path("agent_file_selection_prompt.txt")
+agent_command_prompt_path = Path("agent_command_selection_prompt.txt")
+
+
+class AgentHandler:
+    def __init__(self):
+        self._agent_enabled = False
+
+        self.agent_file_message = ""
+        self.agent_file_selection_prompt = read_prompt(agent_file_selection_prompt_path)
+        self.agent_command_prompt = read_prompt(agent_command_prompt_path)
+
+    # Make this property readonly because we have to set things when we enable agent mode
+    @property
+    def agent_enabled(self):
+        return self._agent_enabled
+
+    def disable_agent_mode(self):
+        self._agent_enabled = False
+
+    async def enable_agent_mode(self):
+        ctx = SESSION_CONTEXT.get()
+
+        self._agent_enabled = True
+        ctx.stream.send(
+            "Finding files to determine how to test changes...", color="cyan"
+        )
+        features = ctx.code_context.get_all_features(CodeMessageLevel.FILE_NAME)
+        messages: List[ChatCompletionMessageParam] = [
+            ChatCompletionSystemMessageParam(
+                role="system", content=self.agent_file_selection_prompt
+            ),
+            ChatCompletionSystemMessageParam(
+                role="system",
+                content="\n".join(
+                    str(feature.path.relative_to(ctx.git_root)) for feature in features
+                ),
+            ),
+        ]
+        model = ctx.config.model
+        response = await ctx.llm_api_handler.call_llm_api(messages, model, False)
+        content = response.choices[0].message.content or ""
+
+        paths = [
+            Path(path) for path in content.strip().split("\n") if Path(path).exists()
+        ]
+        self.agent_file_message = ""
+        for path in paths:
+            file_contents = "\n".join(ctx.code_file_manager.read_file(path))
+            self.agent_file_message += f"{path}\n\n\n{file_contents}"
+
+        ctx.stream.send(
+            "The model has chosen these files to help it determine how to test its"
+            " changes:",
+            color="cyan",
+        )
+        ctx.stream.send("\n".join(str(path) for path in paths))
+        ctx.cost_tracker.log_api_call_stats(
+            prompt_tokens(messages, model),
+            count_tokens(content, model, full_message=False),
+            model,
+        )
+
+        messages.append(
+            ChatCompletionAssistantMessageParam(role="assistant", content=content)
+        )
+        ctx.conversation.add_transcript_message(
```

Comment:
I think we should change the color of this message (since it's a different conversation and only really related to agent mode)